### DISCUSSION TITLE
Calibration Cadence Planning

### DISCUSSION THREAD

**Discussion by kmharrington** [simonsobs/sodetlib#364] [General]
Looking at planning Commissioning Setup and Observations. One of the questions we'll be answering during this time is a lot of cadence / timing questions related to how long different operations should be expected to take and how often we should be expecting to run them. I'd like to get a head start on that by looking back at the previous cooldowns done with full focal planes. I'm particularly thinking of P10r1/2 but any optical cooldowns where all UFM spaces were occupied would be useful. 

Questions I have for folks:
1. How were IVs taken during the cooldown and how often? Ex: Where you running all UFMs and all bias lines at once or serial UFMs with all bias lines or serial UFMs and serial bias lines? 
1a. Were you running with sodetlib default parameters? 
1b. How long did your method take to run? 
1c. How much focal plane heating did you see and how long did it take to cool back down after you ran them? 
1d. Did you bias detectors using IV results? What fraction of detectors were on-transition (20-80% Rn) after biasing?

2. How often were you running `uxm_relock`? Did you have any method of determining it was needed or did you just regularly run it? How long did it take to run each time? and was this done in parallel for all UFMs?

3. How much heating / relaxation time did the focal plane need after overbiasing / biasing up detectors?

**Comment by jlashner** [comment]
I'll think about this / come up with a better response, but to answer I'll probably need to look back at the pysmurf-actions and when each action was run. I remember a while back you sent me a web-page Yilun made on simons1 to display something like pysmurf actions over time for each slot... that would be super helpful to answer this. Is that still running? If not I'll try to make some plots that are similar to that.

**Reply by kmharrington** [reply]
Yeah those were some [g3tsmurf search functions](https://sotodlib.readthedocs.io/en/latest/g3tsmurf.html#file-system-searches) put into a streamlit app. But we'd need to pair that with knowing what you all were doing to run things and reasonings (if any) for doing things one-way versus another.

**Comment by kmharrington** [comment]
I went back through the LATRt UHF cooldown to get some ballparks from timing here. Adding here:
* `uxm_setup` for 1 UFM (not running in parallel): 30 min
* `uxm_relock` for 1 UFM (not running in parallel): 3 min
* IV, serial in bias line: 35 min
* IV, all bias lines: 5 min (~10 min thermal relaxation)
* Bias Steps: 1 min
* Biasing thermal relaxation: ~10 min
* Rebias with bias steps: looks like ~10 min.
